python-frontend/src/main/java/org/sonar/plugins/python/api/symbols/FunctionSymbol.java

```diff
@@ -48,6 +50,9 @@
   @CheckForNull
   LocationInFile definitionLocation();
 
+  @Beta
+  Optional<ClassSymbol> returnTypeHint();
```

Comment:
I think it should return a `Optional<Symbol>`.
In order to be consistent with AST, let's call it `returnTypeAnnotation`.